k8s_test_harness/util/env_util.py

```diff
@@ -0,0 +1,204 @@
+    return val
+
+
+@dataclass
+class RockMetaInfo:
+    """Class representing metadata for a built ROCK as produced by the
+    canonical/k8s-workflows/build_rocks.yaml workflow:
+
+    # TODO(aznashwan): pin exact link upon first stable release of workflows:
+    https://github.com/canonical/k8s-workflows/blob/main/.github/workflows/build_rocks.yaml#L125-L133
+    """
+    name: str
+    version: str
+    rock_dir: str
+    arch: str
+    image: str
+    runs_on: List[str]
+    rockcraft_revision: str
+
+    @classmethod
+    def from_dict(cls, dict_val: Dict[str, str]) -> 'RockMetaInfo':
+        """ Loads fields from the given dict of the form:
+        {
+            "name": rockName,
+            "version": rockVersion,
+            "path": rockPath,
+            "arch": arch,
+            "image": image,
+            "rockcraft-revision": rockcraftRevisions[arch] || '',
+            "runs-on-labels": platformLabels[arch] || [inputs["runs-on"]]
+        }
+
+        :raises ValueError: on missing or null-valued fields.
+        """
+        kwarg_to_key_map = {
+            "name": "name",
+            "version": "version",
+            "rock_dir": "path",
+            "arch": "arch",
+            "image": "image",
+            "rockcraft_revision": "rockcraft-revision",
+            "runs_on": "runs-on-labels"
+        }
+        init_kwargs = {
+            kwarg: dict_val.get(dict_key, None)
+            for kwarg, dict_key in kwarg_to_key_map.items()}
+        unset_keys = [
+            dict_key
+            for kwarg, dict_key in kwarg_to_key_map.items()
+            if init_kwargs[kwarg] == None]
+        if unset_keys:
+            raise ValueError(
+                f"Missing ROCK build meta info fields {unset_keys} from "
+                f"the provided dict: {dict_val}")
+
+        return cls(**init_kwargs)  # pyright: ignore
+
+    @classmethod
+    def from_json_string(cls, json_str: str) -> 'RockMetaInfo':
+        """ Loads fields from JSON string with object of the form:
+        {
+            name: rockName,
+            version: rockVersion,
+            path: rockPath,
+            arch: arch,
+            image: image,
+            "rockcraft-revision": rockcraftRevisions[arch] || '',
+            "runs-on-labels": platformLabels[arch] || [inputs["runs-on"]]
+        }
+
+        :raises json.decoder.JSONDecodeError: on invalid JSON input.
+        :raises ValueError: on missing or null-valued fields.
+        """
+        return cls.from_dict(json.loads(json_str))
+
+    @property
+    def rockfile_path(self) -> str:
+        """Returns the path to the rockfile.yaml file for this image
+        based on the `path` property (RockMetaInfo.rock_dir) of the
+        ROCK's meta info.
+
+        Note that this can be a relative path with respect to the root
+        of the repository the `build_rocks.yaml` workflow had run on.
+        """
+        return os.path.join(self.rock_dir, "rockfile.yaml")
```

Comment:
```suggestion
        return os.path.join(self.rock_dir, "rockcraft.yaml")
```
Isn't this supposed to be `rockcraft.yaml`?

Reply:
Great catch, brain fart, updated the function name for clarity as well!